openmdao/core/problem.py

```diff
@@ -620,6 +622,39 @@ def setup(self, check=True, out_stream=sys.stdout):
 
         return {}
 
+    def _check_solvers(self):
+        """
+        Raise an exception of we detect a LinearGaussSeidel solver if that
+        group has either cycles or states.
+        """
+        for group in self.root.subgroups(recurse=True, include_self=True):
+            if isinstance(group.ln_solver, LinearGaussSeidel) and \
+                                     group.ln_solver.options['maxiter'] == 1:
+                # If group has a cycle and lings can't iterate, that's
+                # an error.
+                graph = group._get_sys_graph()
+                strong = [sorted(s) for s in nx.strongly_connected_components(graph)
+                          if len(s) > 1]
+                if strong:
+                    raise RuntimeError("Group '%s' has a LinearGaussSeidel "
+                                   "solver with maxiter==1 but it contains "
+                                   "cycles %s. To fix this error, change to "
+                                   "a different linear solver, e.g. ScipyGMRES "
+                                   "or PetscKSP, or increase maxiter (only "
+                                   "if you really know what you're doing!)"
```

Comment:
not sure about this warning phrase.. maybe something less dramatic (and judgmental ;)) like "not recommended" or a more helpful "recommended only if ...",  or even just "warning this might result in..."